stestr/scheduler.py

```diff
@@ -108,7 +108,7 @@ def consume_queue(groups):
             for partition in partitions:
                 temp_part = list(partition)
                 random.shuffle(temp_part)
-                out_parts.append(set(temp_part))
+                out_parts.append(list(temp_part))
```

Comment:
oh, I should have mentioned this in the commit message.. This is actually the main reason of this patch. I found that `out_parts.append(set(temp_part))` returns `[{'c_test', 'd_test'}, {'b_test', 'a_test'}]` but `partitions` is `[['c_test', 'd_test'], ['b_test', 'a_test']]` (in python3). So I just fixed it.

```
Traceback (most recent call last):
  File "/Users/igawa/openstack/stestr/stestr/tests/test_scheduler.py", line 64, in test_random_partitions
    sorted_parts, random_parts))
  File "/Users/igawa/openstack/stestr/.tox/py36/lib/python3.6/site-packages/testtools/testcase.py", line 411, in assertEqual
    self.assertThat(observed, matcher, message)
  File "/Users/igawa/openstack/stestr/.tox/py36/lib/python3.6/site-packages/testtools/testcase.py", line 498, in assertThat
    raise mismatch_error
testtools.matchers._impl.MismatchError: !=:
reference = [['c_test', 'd_test'], ['b_test', 'a_test']]
actual    = [{'c_test', 'd_test'}, {'b_test', 'a_test'}]
: The sorted list [['c_test', 'd_test'], ['b_test', 'a_test']] is the same order as the shuffled list [{'c_test', 'd_test'}, {'b_test', 'a_test'}] which is incorrect
```